libs/datasets/timeseries.py

```diff
@@ -1571,3 +1602,42 @@ def make_source_url_tags(ds_in: MultiRegionDataset) -> MultiRegionDataset:
     )
     source_url[TagField.TYPE] = TagType.SOURCE_URL
     return ds_in.append_tag_df(source_url)
+
+
+T_SERIES_OR_DATAFRAME = TypeVar("T_SERIES_OR_DATAFRAME", pd.Series, pd.DataFrame)
+
+# eq=False because instances are large and we want to compare by id instead of value
+@final
+@dataclasses.dataclass(frozen=True, eq=False)
+class MultiRegionDatasetDelta:
+    """Represents a delta/diff between two MultiRegionDataset objects."""
+
+    old: MultiRegionDataset
+    new: MultiRegionDataset
+
+    @staticmethod
+    def make(*, old, new) -> "MultiRegionDatasetDelta":
+        return MultiRegionDatasetDelta(old=old, new=new)
+
+    @property
+    def timeseries_removed(self) -> MultiRegionDataset:
+        """A dataset containing time series, tags and static values in old but not new."""
```

Comment:
What exactly happens with timeseries that are partially filtered via a `start_date`.  At first glance I thought this wouldn't include them, but I guess maybe we add a manual filtered tag/annotation to them and so that gets them included in this result? Is that right?  Might be worth a mention in the comment since it's a bit non-intuitive.

Reply:
You are on the mark. Changes in the set of dates with a real value and changes in the values themselves are ignored but the extra tag will be added in the timeseries_removed dataset. I'd like to incorporate changes within a timeseries more cleanly, not sure how yet.